src/node_url.cc

```diff
@@ -1145,9 +1147,23 @@ namespace url {
           }
           break;
         case kPathStart:
-          state = kPath;
-          if (ch != '/' && !special_back_slash)
-            continue;
+          if (IsSpecial(url->scheme)) {
+            state = kPath;
+            if (ch != '/' && ch != '\\') {
+              continue;
+            }
+          } else if (!has_state_override && ch == '?') {
+            url->query.clear();
```

Comment:
The spec [says](https://url.spec.whatwg.org/#path-start-state) "set _url_’s query to the empty string". A URL's [query](https://url.spec.whatwg.org/#concept-url-query) is initially null. You have to make it clear that the query in this case is empty string rather than undefined by setting `URL_FLAGS_HAS_QUERY` on `url->flags`.